meteor/client/ui/RundownView.tsx

```diff
@@ -2261,10 +2268,13 @@ export const RundownView = translateWithTracker<IProps, IState, ITrackedProps>((
 												rundownId={rundownAndSegments.rundown._id}
 												segmentId={segment._id}
 												playlist={this.props.playlist}
+												rundown={rundownAndSegments.rundown}
 												timeScale={this.state.timeScale}
 												onContextMenu={this.onContextMenu}
 												onSegmentScroll={this.onSegmentScroll}
 												segmentsIdsBefore={rundownAndSegments.segmentIdsBeforeEachSegment[segmentIndex]}
+												rundownIdsBefore={new Set(rundownArray.slice(0, rundownIndex).map((r) => r.rundown._id))}
```

Comment:
This wont be very performant. At the very least this set could be calculated once at the top of the loop on line 2243.
Ideally this is something that should be memoized to help avoid unnecessary re-renders, but I dont know if that will fit into the code well